src/state/slices/swappersSlice/selectors.ts

```diff
@@ -16,9 +22,44 @@ export const selectSellAsset = createDeepEqualOutputSelector(
   swappers => swappers.sellAsset,
 )
 
-export const selectSellAssetAccountId = createSelector(
+// selects the account ID we're selling from
+// note lack of "asset" and "hop" vernacular - this is deliberate
+export const selectSellAccountId = createSelector(
   selectSwappers,
-  swappers => swappers.sellAssetAccountId,
+  selectSellAsset,
+  selectPortfolioAssetAccountBalancesSortedFiat,
+  selectWalletAccountIds,
+  (swappers, sellAsset, accountIdAssetValues, accountIds) => {
+    if (swappers.sellAssetAccountId) return swappers.sellAssetAccountId
```

Comment:
At first glance this is confusing, but I think what we are doing is using `swappers.sellAssetAccountId` as the user's _selected_ `sellAssetAccountId`, and if they haven't selected anything, fallback to the highest, or the first.

A comment to that effect might assist with parsing this code in the future.

Reply:
good idea, fixing